kani-driver/src/concrete_playback.rs

```diff
@@ -41,39 +41,46 @@ impl KaniSession {
-                ensure!(
-                    !self.args.quiet,
-                    "With `--quiet` mode enabled, `--concrete-playback=print` mode can not print test cases."
-                );
-                println!(
-                    "Concrete playback unit test for `{}`:\n```\n{}\n```",
-                    &harness.mangled_name, &concrete_playback.unit_test_str
-                );
-                println!(
-                    "INFO: To automatically add the concrete playback unit test `{}` to the src code, run Kani with `--concrete-playback=InPlace`.",
-                    &concrete_playback.unit_test_name
-                );
-            }
+            if let Some(concrete_vals) = extract_harness_values(processed_items) {
+                let concrete_playback = format_unit_test(&harness.mangled_name, &concrete_vals);
 
-            if *playback_mode == ConcretePlaybackMode::InPlace {
-                if !self.args.quiet {
+                if *playback_mode == ConcretePlaybackMode::Print {
+                    ensure!(
+                        !self.args.quiet,
+                        "With `--quiet` mode enabled, `--concrete-playback=print` mode can not print test cases."
+                    );
                     println!(
-                        "INFO: Now modifying the source code to include the concrete playback unit test `{}`.",
+                        "Concrete playback unit test for `{}`:\n```\n{}\n```",
+                        &harness.pretty_name, &concrete_playback.unit_test_str
+                    );
+                    println!(
+                        "INFO: To automatically add the concrete playback unit test `{}` to the src code, run Kani with `--concrete-playback=InPlace`.",
                         &concrete_playback.unit_test_name
                     );
                 }
-                self.modify_src_code(
-                    &harness.original_file,
-                    harness.original_end_line,
-                    &concrete_playback,
-                )
-                .expect("Failed to modify source code");
+
+                if *playback_mode == ConcretePlaybackMode::InPlace {
+                    if !self.args.quiet {
+                        println!(
+                            "INFO: Now modifying the source code to include the concrete playback unit test `{}`.",
+                            &concrete_playback.unit_test_name
+                        );
+                    }
+                    self.modify_src_code(
+                        &harness.original_file,
+                        harness.original_end_line,
+                        &concrete_playback,
+                    )
+                    .expect("Failed to modify source code");
+                }
+            } else {
+                println!(
+                    concat!(
+                        "WARNING: `{}` failure cannot be reproduced by the concrete playback. ",
```

Comment:
What would be the full message? I was also thinking we should use "panic check" instead of assertion here to align with our documentation. Thoughts?

Reply:
I think it's fair to use the term "panic check". Although we might want to clarify in the documentation that an assertion gets translated down to a panic check, just in case the user doesn't know that.

Full message would be:
```
WARNING: Kani could not produce a concrete playback for `{}` because there were no failing non-panic checks."
```